src/Donation/DonatorSubscriber.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace AppBundle\Donation;
+
+use AppBundle\Repository\AdherentRepository;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class DonatorSubscriber implements EventSubscriberInterface
+{
+    private $adherentRepository;
+
+    public function __construct(AdherentRepository $adherentRepository)
+    {
+        $this->adherentRepository = $adherentRepository;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            DonationEvents::CREATED => ['attachAdherent'],
+        ];
+    }
+
+    /**
+     * Match and attach an adherent with a donator
+     */
+    public function attachAdherent(DonationWasCreatedEvent $donation): void
+    {
+        $donator = $donation->getDonation()->getDonator();
+        $donationRequest = $donation->getDonationRequest();
```

Comment:
why do you need to use donation request here? does Donator entity not have needed data?